### PR TITLE
fix: unify pricing to token-based calculation for all models

### PR DESCRIPTION
**Pull request by cyberprophet**
## Summary
- Replace per-image lookup (ImagePrices/EstimateImageCost) with token-based pricing for image models
- OpenAI API reports input_tokens/output_tokens for image calls — output tokens scale with quality × size
- Add gpt-image-1, gpt-image-1.5, gpt-image-1-mini to unified Prices dictionary with Image modality rates
- Remove ImagePricing record, ImagePrices, EstimateImageCost, ImageKeyComparer
- Bump PricingVersion 2 → 3
- Add 13 unit tests covering text models, image models, cache tokens, case-insensitivity, and unknown model handling

## Pricing verification (2026-04-17)
| Model | Input (per 1M) | Output (per 1M) | Source |
|-------|---------------|-----------------|--------|
| gpt-image-1 | $10.00 | $40.00 | OpenAI pricing page (Image modality) |
| gpt-image-1.5 | $8.00 | $32.00 | OpenAI pricing page (Image modality) |
| gpt-image-1-mini | $2.50 | $8.00 | OpenAI pricing page (Image modality) |

## Test plan
- [x] 13 new unit tests pass (text models, image models, cache tokens, edge cases)
- [ ] Verify P5 consumer compiles with Agency 0.16.0

🤖 Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: agency/Models/ModelPricingTable.cs [hunk 1]
```diff
@@ -73,27 +61,9 @@ public static class ModelPricingTable
...
     public static decimal? EstimateCost(ApiUsageEvent usage)
     {
-        if (usage.ImageQuality is not null || usage.ImageSize is not null)
-            return EstimateImageCost(usage.Model, usage.ImageQuality, usage.ImageSize);
-
-        if (usage.Model.StartsWith("gpt-image", StringComparison.OrdinalIgnoreCase)
-            || string.Equals(usage.Purpose, "image", StringComparison.OrdinalIgnoreCase))
-            return EstimateImageCost(usage.Model, null, null);
-
         return EstimateCost(usage.Provider, usage.Model, usage.InputTokens, usage.OutputTokens);
```
Comment (MintSurf) [inline agency/Models/ModelPricingTable.cs:74]: **<sub><sub>P1 Badge</sub></sub>  Missing fallback when image usage is absent**

The old implementation charged image events from `ImageQuality`/`ImageSize` whenever we knew this was an image request. This line now always prices from `InputTokens`/`OutputTokens`, but the image producers still build `ApiUsageEvent` with `(usage?.InputTokenCount ?? 0)` and `(usage?.OutputTokenCount ?? 0)`, which means a null or incomplete OpenAI `Usage` object turns into a `$0` charge instead of the previous fixed image price. Because this affects billing/data integrity, we should either preserve the old fallback path for image events or fail closed instead of silently undercounting cost.

Useful? React with 👍 / 👎.

**Review by MintSurf — CHANGES_REQUESTED** [review]
## Review Summary
- Outcome: request-changes
- Summary: Moving image pricing to token-based estimation makes sense, but this patch removes the existing fallback for image events when the OpenAI SDK does not return usage details. In that case the new estimator records a zero-dollar cost instead of the prior quality/size-based price, which is a billing regression.
- Checks: Skipped project checks because the repository does not contain a *.sln, package.json, or pubspec.yaml under the requested rules.

File: agency/Models/ModelPricingTable.cs [hunk 2]
```diff
@@ -73,27 +61,9 @@ public static class ModelPricingTable
...
-
-        return ImagePrices.TryGetValue((model, quality, size), out var pricing)
-            ? pricing.CostPerImage
-            : null;
-    }
-
-    /// <summary>Unified estimator: routes to <see cref="EstimateCost"/> for text models or <see cref="EstimateImageCost"/> for image models based on <see cref="ApiUsageEvent"/>.</summary>
+    /// <summary>Unified estimator: resolves provider+model from <see cref="ApiUsageEvent"/> and delegates to the token-based calculator.</summary>
     public static decimal? EstimateCost(ApiUsageEvent usage)
     {
```
Comment (cyberchacha) [inline agency/Models/ModelPricingTable.cs:69]: **<sub><sub>P3 Badge</sub></sub>  Vestigial `ImageQuality`/`ImageSize` on `ApiUsageEvent`**

`ApiUsageEvent` still carries `ImageQuality` and `ImageSize` fields, and callers like `GptService.Image.cs` still populate them, but this method no longer uses them for pricing. Consider a follow-up to either:
1. Remove the fields from `ApiUsageEvent` if they serve no other purpose (logging, telemetry), or
2. Keep them explicitly for observability and document that they are not pricing inputs.

Not blocking — the current code is correct.

Useful? React with 👍 / 👎.

**Review by cyberchacha — DISMISSED** [review]
## Review Summary
- Outcome: approve
- Summary: Clean simplification that unifies image and text pricing under a single token-based calculation. Well-tested with 13 unit tests covering all key scenarios. The approach correctly leverages OpenAI's token reporting for image calls.
